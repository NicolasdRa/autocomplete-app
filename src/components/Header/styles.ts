import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
	createStyles({
		root: {
			flexGrow: 1,
		},

		menuButton: {
			marginRight: theme.spacing(2),
		},
		title: {
			flexGrow: 1,
		},

		logo: {
			width: '2rem',
			marginRight: '1rem',
		},
	}),
);
