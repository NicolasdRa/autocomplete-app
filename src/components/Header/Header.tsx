import {
	AppBar,
	Box,
	Toolbar,
	Typography,
	IconButton,
} from '@material-ui/core';
import AccountCircle from '@material-ui/icons/AccountCircle';
import logo from '../../assets/logo-rounded.png';

import { useStyles } from './styles';

export const Header: React.FC = () => {
	const classes = useStyles();

	return (
		<div className={classes.root}>
			<AppBar position='static'>
				<Toolbar>
					<Box>
						<a href='!#' target='_blank' rel='noreferrer'>
							<img
								className={classes.logo}
								src={logo}
								alt='logo'
							/>
						</a>
					</Box>
					<Typography variant='h6' className={classes.title}>
						Autocomplete App
					</Typography>
					<IconButton
						aria-label='account of current user'
						// aria-controls='menu-appbar'
						// aria-haspopup='false'
						onClick={() => console.log('clicked')}
						color='inherit'>
						<AccountCircle />
					</IconButton>
				</Toolbar>
			</AppBar>
		</div>
	);
};
