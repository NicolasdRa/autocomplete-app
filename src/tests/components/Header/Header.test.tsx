import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { Header } from '../../../components/Header/Header';

describe('tests on <Header/>', () => {
	const wrapper = mount(<Header />);

	test('should render <Header/> correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
