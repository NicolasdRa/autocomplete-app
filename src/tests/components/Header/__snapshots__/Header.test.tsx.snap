// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <Header/> should render <Header/> correctly 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <div
      className="MuiBox-root MuiBox-root-3"
    >
      <a
        href="!#"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="logo"
          className="makeStyles-logo-1"
          src="logo-rounded.png"
        />
      </a>
    </div>
    <h6
      className="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
    >
      Autocomplete App
    </h6>
    <button
      aria-label="account of current user"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</header>
`;
